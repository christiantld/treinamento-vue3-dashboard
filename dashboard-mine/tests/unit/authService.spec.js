/* eslint-disable space-before-function-paren */
import mockAxios from 'axios'
import AuthService from '@/services/auth.js'

jest.mock('axios')

describe('Auth Service', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('It should return a token when user login', async () => {
    const token = '123.123.123'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({
        data: { token }
      })
    })
    const response = await AuthService(mockAxios).login({
      email: 'igor@igor.me',
      password: '1234'
    })
    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('It should return an user when user register', async () => {
    const user = {
      name: 'Igor',
      email: 'igor@igor.me',
      password: '1234'
    }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { user } })
    })
    const response = await AuthService(mockAxios).register(user)

    expect(response.data).toHaveProperty('user')
    expect(response.data.user).toHaveProperty('name')
    expect(response.data.user).toHaveProperty('password')
    expect(response.data.user).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('It should throw an error when not found', async () => {
    const errors = { status: 404, statusText: 'Not found' }

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })
    const response = await AuthService(mockAxios).login({
      email: 'igor@igor.me',
      password: '1234'
    })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
})
