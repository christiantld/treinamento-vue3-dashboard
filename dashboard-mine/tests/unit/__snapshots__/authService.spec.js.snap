// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Service It should return a token when user login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`Auth Service It should return an user when user register 1`] = `
Object {
  "data": Object {
    "user": Object {
      "email": "igor@igor.me",
      "name": "Igor",
      "password": "1234",
    },
  },
  "errors": null,
}
`;
